resources/functions/user-management/index.py

```diff
@@ -8,67 +8,55 @@
 from aws_lambda_powertools import Logger
 from aws_lambda_powertools.logging import correlation_paths
 from aws_lambda_powertools.event_handler import APIGatewayRestResolver
-import idp_object_factory
+from cognito_user_management_service import CognitoUserManagementService
 
 tracer = Tracer()
 logger = Logger()
 app = APIGatewayRestResolver()
 
 idp_name = os.environ['IDP_NAME']
 idp_details=json.loads(os.environ['IDP_DETAILS'])
-
-idp_user_mgmt_service = idp_object_factory.get_idp_user_mgmt_object(idp_name)
+idp_user_mgmt_service = CognitoUserManagementService()
 
 @app.post("/users")
 @tracer.capture_method
 def create_user():
     user_details = app.current_event.json_body
-    logger.info("Request received to create new user")
     user_details['idpDetails'] = idp_details
     response = idp_user_mgmt_service.create_user(user_details)
-    logger.info("Request completed to create new user ")
-
+    logger.info(response)
     return utils.create_success_response("New user created")
 
 @app.get("/users")
 @tracer.capture_method
 def get_users():
-    users = []
     user_details = {}
-    user_details['idpDetails'] = idp_details  
-    
-    logger.info("Request received to get user")
+    user_details['idpDetails'] = idp_details
     response = idp_user_mgmt_service.get_users(user_details)
-        
-    logger.info(response) 
-    num_of_users = len(response)
+    logger.info(response)
     return utils.generate_response(response)
 
-
 @app.get("/users/<username>")
 @tracer.capture_method
 def get_user(username):
     user_details = {}
     user_details['idpDetails'] = idp_details
     user_details['userName'] = username
+    response = idp_user_mgmt_service.get_user(user_details)
+    if (response == None):
+        logger.info("User not found")
+        return utils.create_not_found_response("User not found")
+    logger.info(response)
+    return utils.generate_response(response)
```

Comment:
have we considered just using what lambda powertools provides us directly to return a response? Something like this:

https://github.com/awslabs/sbt-aws/blob/2463b9a751223aca90b6e4d95df374ca11342e97/resources/functions/tenant-config/index.py#L55

There's an example in the snippet in this section: https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#event-resolvers

Reply:
I can refactor this as well, but I thought it would be best to refactor this part together with making the APIs RESTful and standardizing verbs / response codes and response bodies.

Reply:
done